### PR TITLE
Fix texturespecification/teximage2d_params.html

### PR DESCRIPTION
**Pull request by zhenyao**

### CODE REVIEW

File: sdk/tests/deqp/framework/opengl/simplereference/sglrReferenceContext.js [hunk 1]
```diff
@@ -4148,14 +4148,16 @@ goog.scope(function() {
                 texture.allocLevel(level, storageFmt, width, height);
 
             if (data) {
-                var rowPitch = deMath.deAlign32(width * transferFmt.getPixelSize(), this.m_pixelUnpackAlignment);
+                var rowLen = this.m_pixelUnpackRowLength > 0 ? this.m_pixelUnpackRowLength : width;
+                var rowPitch = deMath.deAlign32(rowLen * transferFmt.getPixelSize(), this.m_pixelUnpackAlignment);
+                var skip = this.m_pixelUnpackSkipRows * rowPitch + this.m_pixelUnpackSkipPixels * transferFmt.getPixelSize();
                 src = new tcuTexture.ConstPixelBufferAccess({
```
Comment (zhenyao) [inline sdk/tests/deqp/framework/opengl/simplereference/sglrReferenceContext.js]: c++ code: https://code.google.com/p/chromium/codesearch#chromium/src/third_party/deqp/src/framework/opengl/simplereference/sglrReferenceContext.cpp&l=929

File: sdk/tests/deqp/functional/gles3/es3fTextureSpecificationTests.js [hunk 2]
```diff
@@ -1677,7 +1677,7 @@ goog.scope(function() {
         );
 
         // Fill data with grid.
-        data = new ArrayBuffer(rowPitch * height);
+        data = new ArrayBuffer(rowPitch * height + this.m_skipPixels * pixelSize);
```
Comment (zhenyao) [inline sdk/tests/deqp/functional/gles3/es3fTextureSpecificationTests.js]: This is a bug in the original c++ code. Without fixing this, the skippixels test still pass, but if we test with a different skippixels value (say a large number like 100), then it fails with this fix.

File: sdk/tests/deqp/functional/gles3/es3fTextureSpecificationTests.js [hunk 3]
```diff
@@ -1694,16 +1694,23 @@ goog.scope(function() {
...
-        tcuTextureUtil.fillWithGrid(access, 4, colorA, colorB
+        tcuTextureUtil.fillWithGrid(accessWithOffset, 4, colorA, colorB
         );
+        var access = new tcuTexture.PixelBufferAccess({
+                format: this.m_texFormat,
+                width: this.m_width,
+                height: this.m_height,
+                rowPitch: rowPitch,
+                data: data,
+                });
```
Comment (zhenyao) [inline sdk/tests/deqp/functional/gles3/es3fTextureSpecificationTests.js]: c++ code: https://code.google.com/p/chromium/codesearch#chromium/src/third_party/deqp/src/modules/gles3/functional/es3fTextureSpecificationTests.cpp&l=1107

**Comment by zhenyao** [comment]
@kenrussell Please review.

**Comment by kenrussell** [comment]
Great work Mo tracking down these bugs. LGTM.

**Pull request merged**